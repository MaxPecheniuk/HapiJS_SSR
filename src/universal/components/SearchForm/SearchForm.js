//@flow
import React, {Fragment} from 'react';
import { InputText } from '../share.components/inputText/inputText';
import { searchFormActionCreators } from './actionCreators/searchForm.actionCreators';
import { connect } from 'react-redux';

import './searchForm.scss';
import { withRouter } from 'react-router';

type SearchFormState = {
  inputValue: string
}
type Props = {
  dispatch: any
}

class SearchForm extends React.Component<Props, SearchFormState> {
  state = {
    inputValue: '',
    redirect: false
  }

  onChange = (event: SyntheticEvent<HTMLInputElement>) => {
    this.setState({
      inputValue: event.currentTarget.value
    })
  }

  onSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
    event.preventDefault();
    this.props.history.push({search: this.state.inputValue});
    this.props.dispatch(searchFormActionCreators(this.state.inputValue))


  }

  render() {
    // console.log(this.state.redirect);
    // if (this.state.redirect){
    console.log(this.props);
    //   return  <Redirect to={"/home"} />
    // }
    return (
      <Fragment>

        <form onSubmit={this.onSubmit} className="search-form">
          <InputText
            value={this.state.inputValue}
            placeholder="Search title"
            onChange={this.onChange}
            name="inputValue"
            className="search-form__input-field"
          />
          <button
            onSubmit={this.onSubmit}
            className="search-form__submit-btn"
          >Submit</button>
        </form>
      </Fragment>
    )
  }
}


export default withRouter(connect()(SearchForm))