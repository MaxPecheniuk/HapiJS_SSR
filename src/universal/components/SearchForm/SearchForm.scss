@import "../../styles/variables";

.search-form {
  display: flex;
  width: 45%;
  margin-right: 50px;
  &__input-field{
    background-color: $backgroundLight;
    border: none;
    padding: 10px;
    box-shadow: 0px 10px 30px -10px rgba(0, 0, 0, .3);
    font-size: 1.25rem;
    width: 90%;
    height: 35px;
    vertical-align: middle;
    border-bottom-left-radius: 5px;
    border-top-left-radius: 5px;
    &:focus {
      outline: none;
      &::placeholder {
        color: transparent;
      }
    }
  }
  &__submit-btn{
    height: 35px;
    border:1px solid $grey;
    background: $grey;
    color: $backgroundLight;
    border-bottom-right-radius: 5px;
    border-top-right-radius: 5px;
  }
}
@media screen and (max-width: 768px) {
  .search-form {
    width: 80%
  }
  .post-item {
    width: 80%
  }
}
@media screen and (max-width: 540px){
  .search-form {
    margin-left: 10px;
  }


}
