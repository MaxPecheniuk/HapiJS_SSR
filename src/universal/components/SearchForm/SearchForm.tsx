import * as React from 'react';
import { Fragment, SyntheticEvent, useState } from 'react';
import { InputText } from '../share.components/inputText/InputText';
import { RouteComponentProps, withRouter } from 'react-router';
import { FormattedMessage } from 'react-intl';
import * as queryString from 'querystring';
import { History, Location } from 'history';

import './SearchForm.scss';

interface ISearchFormProps {
  history?: History;
  location?: Location;
}

const SearchForm: React.FunctionComponent<ISearchFormProps & RouteComponentProps<{}>> =
  (props: ISearchFormProps & RouteComponentProps<{}>) => {
    const [inputValue, inputValueOnChange] = useState<string>('');

    const onChange = (event: SyntheticEvent<HTMLInputElement>) => {
      inputValueOnChange(event.currentTarget.value);
    };

    const onSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
      event.preventDefault();

      let parsed = queryString.parse(props.location.search.replace('?', ''));
      if (Object.keys(parsed).length > 0) {
        parsed['title'] = inputValue;
      } else { parsed.search = inputValue; }
      const stringified = queryString.stringify(parsed);
      props.history.push({pathname: '/', search: stringified});
    };

    return (
      <Fragment>
        <form onSubmit={onSubmit} className="search-form">
          <FormattedMessage id="form.inputPlaceholder">
            {placeholder => (<InputText
              value={inputValue}
              placeholder={placeholder}
              onChange={onChange}
              name="inputValue"
              className="search-form__input-field"
            />)}
          </FormattedMessage>

          <button
            type="submit"
            className="search-form__submit-btn"
          >
            <FormattedMessage id="form.submitBtn"/>
          </button>
        </form>
      </Fragment>
    );
  };

export default withRouter(SearchForm);