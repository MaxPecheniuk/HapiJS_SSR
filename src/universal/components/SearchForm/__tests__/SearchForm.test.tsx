import * as React from 'react';
import * as enzyme from 'enzyme';
import SearchForm from '../SearchForm';

describe('LanguageToggle component', () => {
  let component;
  beforeEach(() => {
    component = enzyme.shallow(<SearchForm/>);
  })

  it('should render correctly with no props', () => {
    expect(component).toMatchSnapshot();
  });

  it('shallow wrapper instance should be null', () => {
    const instance = component.instance().onChange('ddd');
    expect(instance).toEqual(component.state('inputValue').toEqual('ddd'));
  });
});