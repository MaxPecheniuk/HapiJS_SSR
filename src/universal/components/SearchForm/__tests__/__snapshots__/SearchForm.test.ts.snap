// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageToggle component should render correctly with no props 1`] = `ShallowWrapper {}`;
